cpp/src/lance/arrow/type.cc

```diff
@@ -161,16 +167,34 @@ ::arrow::Result<std::shared_ptr<::arrow::DataType>> FromLogicalType(
     return ::arrow::fixed_size_binary(size);
   }
 
-  if (logical_type.starts_with("dict")) {
+  if (logical_type.starts_with("extension")) {
```

Comment:
Should we check `starts_with("extension:")` (with colon) instead of `extension`.

Reply:
If we want to change this we should also change the other checks for consistency. Also if we want to change the separator to account for stuff like parametric types?

Reply:
i would say let’s do this separately since we should change all of the other type checks for consistency.